src/components/SearchPage.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useEffect } from 'react'
+import Card from './Card'
+import Pagination from './Pagination'
+import { isCollected } from 'utils'
+
+type SearchPageProps = {
+  collection: Collection
+  dispatch: React.Dispatch<Action>
+}
+
+const SearchPage = ({ collection, dispatch }: SearchPageProps) => {
+  const { cards } = collection
+
+  const [formData, setFormData] = useState({
+    pokemonName: '',
+  })
+  const [responseData, setResponseData] = useState<ResponseDataType>({
+    data: [],
+    page: 1,
+    pageSize: 0,
+    count: 1,
+    totalCount: 0,
+  })
+  const [query, setQuery] = useState('')
+  const [currentPage, setCurrentPage] = useState(1)
+  const [isLoaded, setIsLoaded] = useState(false)
+  const urlEndpoint = `https://api.pokemontcg.io/v2/cards?pageSize=48&page=${currentPage}&q=${query}`
+
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target
+    setFormData((prevFormData) => {
+      return {
+        ...prevFormData,
+        [name]: value,
+      }
+    })
+  }
+
+  const setPage = (page: number) => {
```

Comment:
same thing here, this is a one-line arrow function so no need to seperate it out